### ISSUE TITLE
The `guard` macro is broken

### ISSUE THREAD

**Issue by laarc** [sctb/lumen#13]
My understanding of the guard macro is that `(guard 42)` should result in `(true 42)`.  Indeed, this is what `test.l` seems to test for:

```
(test= '(true 42) (guard 42))
```

However, run `bin/lumen` and type:

```
> (guard 42)
()
```

Also, 
```
$ bin/lumen -e '(guard 42)'
()
```

The problem is due to how the guard macro is implemented:

```
> (macroexpand '(guard 42))
("do" ("%local" "_|_e,_x1|" ("xpcall" ("%function" () ("do" 42)) "%message-handler")) ("do" ("do" ("%array" "_e" "_x1"))))

> (print (compile (macroexpand '(guard 42))))
local _124_e144_x4124 = xpcall(function ()
  42
end, _37message_handler)
{_e1, _x4}
```

I fixed the guard macro in my most recent pull request.  But if the guard macro is valid as written, then I should track down where the difference in behavior is.

To be clear, the tests work correctly:

```
> (define-test guard (test= '(true 42) (guard 42)))
> (run)
 1 passed, 0 failed
> (define-test guard (test= nil (print (string (guard 42)))))
> (run)
(true 42)
 3 passed, 0 failed
> 
```

But the REPL doesn't, nor does eval:

```
> (eval '(guard 42))
()
```

**Comment by sctb** [comment]
Thanks! I see how the pull request implements the fix—I'll adapt it to the current `(guard expr)` implementation.

**Comment by laarc** [comment]
Thanks a lot!  Fixed in 2847da6.

**Comment by sctb** [comment]
I actually cherry-picked your commit, and thought I committed that before 2847da6, but I messed it up. Sorry!
